rspace/src/main/scala/coop/rchain/rspace/HotStore.scala

```diff
@@ -167,17 +167,19 @@ private class InMemHotStore[F[_]: Sync, C, P, A, K](
   def putJoin(channel: C, join: Seq[C]): F[Unit] =
     for {
       joins <- getJoins(channel)
-      _ <- S.flatModify { cache =>
+      _ <- if (!joins.contains(join)) S.flatModify { cache =>
```

Comment:
thought: would the order of `join: Seq[C]` affect this check?

Reply:
You're probably right, but this would mean that the current `IStore` implementations are also affected